src/Toggl/Common/TogglClientAuthPlugin.php

```diff
@@ -72,10 +71,15 @@ public function onRequestBeforeSend(Event $event)
      * authentication scheme becomes more complex. Separating it out from the
      * event handler allows us to test this code more easily.
      *
-     * @param \Guzzle\Http\Message\Request $request
+     * @param \GuzzleHttp\Psr7\Request $request
+     *
+     * @return \GuzzleHttp\Psr7\MessageTrait $trait
      */
     public function setAuth(Request $request)
     {
-        $request->setAuth($this->username, $this->password);
+        return $request->withHeader(
+          'Authorization',
+          'Basic ' . base64_encode($this->username . ':' . $this->password)
+        );
```

Comment:
Is this plugin class even necessary in Guzzle 6? I thought 'auth' was passed in the config array to the client class?

Reply:
Very good point. I'll look into removing this class and moving the auth into the `Client` config.